File: Java/AlgorithmsPractice/src/findMissing/FindMissingNumber.java
package findMissing;

public class FindMissingNumber {

    private int[] arrayOfNumbers;

    public int getResult() {
        if (this.arrayOfNumbers.length < 1) {
            return 1;
        }
        int n = arrayOfNumbers.length + 1;
        int expectedSumOfNumbers = (n * (n + 1) / 2);
        int actualSum = 0;
        for (int arrayOfNumber : arrayOfNumbers) {
            actualSum += arrayOfNumber;
        }
        return expectedSumOfNumbers - actualSum;
    }

    public FindMissingNumber(int[] arrayOfNumbers) {
        this.arrayOfNumbers = arrayOfNumbers;
    }
}

File: Kotlin/AlgorithmsPractice/src/findMissing/FindMissingNumber.kt
package findMissing

// note: this only works for arrays with '1' as the lowest value
class FindMissingNumber(
    private val arrayOfNumbers: IntArray
) {
    fun getResult(): Int {
        if(arrayOfNumbers.isEmpty()) {
            return 1
        }
        val n: Int = arrayOfNumbers.size + 1
        val expectedSum: Int = (n * (n + 1) / 2)
        var actualSum: Int = 0;
        for (number in arrayOfNumbers) {
            actualSum += number;
        }
        return expectedSum - actualSum
    }
}